presto-main/src/main/java/com/facebook/presto/operator/AbstractOperatorSpillContext.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator;
+
+import com.facebook.presto.spiller.LocalSpillContext;
+
+public abstract class AbstractOperatorSpillContext
```

Comment:
I refactored it.
There is now `SpillContext`. Concrete classes are not used around the codebase any more. I also unified interface of `OperatorSpillContext` and `LocalSpillContext`. 
Changes allowed to get rid of passing `Supplier<LocalSpillContext>` and just use `SpillContext` in that place in one of following commits.